### PR TITLE
[Drools 5066]First step to isolate Kafka deps

### PR DESCRIPTION
**Pull request by desmax74**
This is the first step of the drools-4706

### CODE REVIEW

**Comment by desmax74** [comment]
@jiripetrlik @mariofusco @danielezonca

File: hacep/src/main/java/org/kie/hacep/EnvConfig.java [hunk 1]
```diff
@@ -146,71 +148,54 @@ public EnvConfig withMaxSnapshotRequestAttempts(String maxSnapshotRequestAttempt
...
                 this.pollDuration = Duration.ofMillis(pollTimeout);
                 break;
-            case sec:
+            case SECONDS:
                 this.pollUnit = PollUnit.SECOND;
                 this.pollDuration = Duration.ofSeconds(pollTimeout);
                 break;
             default:
-                throw new RuntimeException("No pollTimeUnit provided");
+                throw new ConfigurationException("No pollTimeUnit provided");
```
[the remark this answers is not in the document]
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/EnvConfig.java]: Sonarcloud ask a more specific exception when RuntimeExceptions is used

File: hacep/src/main/java/org/kie/hacep/EnvConfig.java [hunk 2]
```diff
@@ -146,71 +148,54 @@ public EnvConfig withMaxSnapshotRequestAttempts(String maxSnapshotRequestAttempt
...
         }
         return this;
     }
 
-    public EnvConfig withPollSnapshotTimeUnit(String pollSnapshotTimeUnit){
-        switch (pollSnapshotTimeUnit){
-            case millisec:
+    public EnvConfig withPollSnapshotTimeUnit(String pollSnapshotTimeUnit) {
+        if(pollSnapshotTimeUnit == null){
+            throw new ConfigurationException("No pollSnapshotTimeUnit provided");
```
[the remark this answers is not in the document]
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/EnvConfig.java]: Sonarcloud ask a more specific exception when RuntimeExceptions is used

File: hacep/src/main/java/org/kie/hacep/exceptions/ConfigurationException.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.hacep.exceptions;
+
+public class ConfigurationException extends RuntimeException {
```
[the remark this answers is not in the document]
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/exceptions/ConfigurationException.java]: Sonarcloud ask a more specific exception when RuntimeExceptions is used

File: integration/src/test/java/org/kie/hacep/KafkaUtils.java [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+import org.kie.remote.impl.RemoteKieSessionImpl;
+import org.kie.remote.impl.RemoteStreamingKieSessionImpl;
+import org.kie.remote.impl.consumer.Listener;
+import org.kie.remote.impl.producer.Sender;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KafkaUtils implements AutoCloseable {
+
+    private static final String ZOOKEEPER_HOST = "127.0.0.1";
```
[the remark this answers is not in the document]
Comment (desmax74) [reply integration/src/test/java/org/kie/hacep/KafkaUtils.java]: Is the address of the local Kafka used for tests

File: hacep/src/main/java/org/kie/hacep/EnvConfig.java [hunk 5]
```diff
@@ -146,71 +148,54 @@ public EnvConfig withMaxSnapshotRequestAttempts(String maxSnapshotRequestAttempt
...
                 this.pollSnapshotDuration = Duration.ofMillis(pollSnapshotTimeout);
                 break;
-            case sec:
+            case SECONDS:
                 this.pollUnitSnapshot = PollUnit.SECOND;
                 this.pollSnapshotDuration = Duration.ofSeconds(pollSnapshotTimeout);
                 break;
             default:
-                throw new RuntimeException("No pollSnapshotTimeUnit provided");
+                throw new ConfigurationException("No pollSnapshotTimeUnit provided");
```
[the remark this answers is not in the document]
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/EnvConfig.java]: Sonarcloud ask a more specific exception when RuntimeExceptions is used

File: hacep/src/main/java/org/kie/hacep/EnvConfig.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiripetrlik) [reply hacep/src/main/java/org/kie/hacep/EnvConfig.java]: Ok, understand. So, let's go with new exceptions!

File: hacep/src/main/java/org/kie/hacep/core/infra/DefaultSessionSnapShooter.java [hunk 6]
```diff
@@ -60,8 +58,7 @@ public void serialize(KieSessionContext kieSessionContext, String lastInsertedEv
         try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
             EventProducer<byte[]> producer = new EventProducer<>();
             producer.start(Config.getSnapshotProducerConfig());
-            marshallers.newMarshaller(kieSessionContext.getKieSession().getKieBase()).marshall(out,
-                                                                                               kieSessionContext.getKieSession());
+            marshallers.newMarshaller(kieSessionContext.getKieSession().getKieBase()).marshall(out, kieSessionContext.getKieSession());
```
[the remark this answers is not in the document]
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/core/infra/DefaultSessionSnapShooter.java]: ok

File: hacep/src/main/java/org/kie/hacep/core/infra/consumer/LocalConsumer.java [hunk 7]
```diff
@@ -25,40 +25,39 @@
...
             } else {
                 break;
             }
         }
     }
 
     @Override
-    public void stop() { }
+    public void stop() {
+        //do nothing
```
[the remark this answers is not in the document]
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/core/infra/consumer/LocalConsumer.java]: ok

File: integration/src/test/java/org/kie/hacep/RemoteStreamingKieSessionImplTest.java [hunk 8]
```diff
@@ -16,30 +16,34 @@
...
     public void getFactCountTest() throws Exception {
+        Properties props = getTestProperties();
         Bootstrap.startEngine(envConfig);
         Bootstrap.getConsumerController().getCallback().updateStatus(State.LEADER);
-        kafkaServerTest.insertBatchStockTicketEvent(7,
-                                                    topicsConfig,
-                                                    RemoteStreamingKieSession.class);
+        kafkaServerTest.insertBatchStockTicketEvent(7, topicsConfig, RemoteStreamingKieSession.class, InfraFactory.getListener(props, false));
+        Producer prod = InfraFactory.getProducer(false);
         RemoteStreamingKieSessionImpl client = new RemoteStreamingKieSessionImpl(Config.getProducerConfig("FactCountConsumerTest"),
```
[the remark this answers is not in the document]
Comment (desmax74) [reply integration/src/test/java/org/kie/hacep/RemoteStreamingKieSessionImplTest.java]: Ok

File: hacep/src/main/java/org/kie/hacep/consumer/CommandHandler.java [hunk 9]
```diff
@@ -261,22 +253,17 @@ public void visit(SnapshotOnDemandCommand command) {
...
         if (ks != null) {
             ReleaseId releaseId = ks.newReleaseId(command.getGroupID(), command.getArtifactID(), command.getVersion());
-            if(envConfig.isUpdatableKJar()) {
+            if (envConfig.isUpdatableKJar()) {
                 try {
                     kieSessionContext.getKieContainer().updateToVersion(releaseId);
-                    msg = new UpdateKjarMessage(command.getId(), Boolean.TRUE);
+                    msg = new UpdateKJarMessage(command.getId(), Boolean.TRUE);
                 } catch (java.lang.UnsupportedOperationException ex) {
                     logger.info("It isn't possible update a classpath container to a new version");
```
[the remark this answers is not in the document]
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/consumer/CommandHandler.java]: Ok

File: ha-core-infra/src/main/java/org/kie/remote/impl/consumer/KafkaListenerThread.java [hunk 10]
```diff
@@ -39,55 +38,39 @@
...
-            consumer.seek(part, lastOffset - 1);
-        }
+    public void init(Map<String, CompletableFuture<Object>> requestsStore) {
+        this.requestsStore = requestsStore;
     }
 
     @Override
     public void run() {
+        if (requestsStore == null) {
+            throw new InitializeException("Request store not initialized");
```
[the remark this answers is not in the document]
Comment (desmax74) [reply ha-core-infra/src/main/java/org/kie/remote/impl/consumer/KafkaListenerThread.java]: Ok

File: hacep/src/test/java/org/kie/hacep/exceptions/ExceptionsTest.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.hacep.exceptions;
+
+import org.junit.Test;
+
+public class ExceptionsTest {
```
Comment (danielezonca) [inline hacep/src/test/java/org/kie/hacep/exceptions/ExceptionsTest.java]: Ok please remove this file :smile:

File: hacep/src/test/java/org/kie/hacep/consumer/BidirectionalMapTest.java [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+package org.kie.hacep.consumer;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class BidirectionalMapTest {
+
+    @Test
+    public void bidirectionalMapTest(){
```
Comment (danielezonca) [inline hacep/src/test/java/org/kie/hacep/consumer/BidirectionalMapTest.java]: Can you please test corner cases in addition to the happy path: what if the key does not exist etc

File: kie-remote/src/main/java/org/kie/remote/command/EventInsertCommand.java [hunk 13]
```diff
@@ -18,27 +18,30 @@
...
+public class EventInsertCommand extends WorkingMemoryActionCommand implements VisitableCommand, Serializable {
 
-    private Object object;
+    private Serializable object;
 
-    public EventInsertCommand(){}
+    public EventInsertCommand() { /*For serialization*/}
 
-    public EventInsertCommand( Object object, String entryPoint ) {
+    public EventInsertCommand(Object object, String entryPoint) {
```
Comment (danielezonca) [inline kie-remote/src/main/java/org/kie/remote/command/EventInsertCommand.java]: I think it is better tha change object type to `Serializable` instead of just do an internal cast

File: hacep/src/test/java/org/kie/hacep/exceptions/ExceptionsTest.java (hunk 11 above)
Comment (desmax74) [reply hacep/src/test/java/org/kie/hacep/exceptions/ExceptionsTest.java]: You know, sonarcloud complains about classes without test :D

Comment (danielezonca) [reply hacep/src/test/java/org/kie/hacep/exceptions/ExceptionsTest.java]: I know but let's ignore sonarcloud for this :)
Btw you can also add a property in the pom to exclude some classes from coverage (`jacoco.excludes`)

File: hacep/src/main/java/org/kie/hacep/exceptions/ConfigurationException.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/exceptions/ConfigurationException.java]: Ok, let's see if sonarcloud complains

File: ha-core-infra/src/main/java/org/kie/remote/impl/consumer/InfraFactory.java [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+import org.kie.remote.impl.consumer.ListenerThread;
+import org.kie.remote.impl.consumer.LocalListenerThread;
+import org.kie.remote.impl.producer.EventProducer;
+import org.kie.remote.impl.producer.LocalProducer;
+import org.kie.remote.impl.producer.Producer;
+
+import static org.kie.remote.CommonConfig.LOCAL_MESSAGE_SYSTEM_CONF;
+import static org.kie.remote.util.ConfigurationUtil.readBoolean;
+
+public class InfraFactory {
```
[the remark this answers is not in the document]
Comment (desmax74) [reply ha-core-infra/src/main/java/org/kie/remote/impl/consumer/InfraFactory.java]: ok

File: hacep/src/test/java/org/kie/hacep/consumer/BidirectionalMapTest.java (hunk 12 above)
Comment (desmax74) [reply hacep/src/test/java/org/kie/hacep/consumer/BidirectionalMapTest.java]: Ok

File: kie-remote/src/main/java/org/kie/remote/command/EventInsertCommand.java (hunk 13 above)
Comment (desmax74) [reply kie-remote/src/main/java/org/kie/remote/command/EventInsertCommand.java]: ok, going to change the stack of calls

File: hacep/src/main/java/org/kie/hacep/exceptions/UnsupportedStateException.java [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.hacep.exceptions;
+
+public class UnsupportedStateException extends RuntimeException {
```
[the remark this answers is not in the document]
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/exceptions/UnsupportedStateException.java]: ok

File: hacep/src/main/java/org/kie/hacep/consumer/FactHandlesManager.java [hunk 16]
```diff
@@ -51,33 +52,33 @@ public FactHandlesManager(KieSession kieSession) {
...
     }
 
     public FactHandle mapRemoteFactHandle(RemoteFactHandle remoteFH) {
-        return fhMap.computeIfAbsent( remoteFH, this::getFactHandleById );
+        return fhMap.computeIfAbsent(remoteFH, this::getFactHandleById);
     }
 
-    private InternalFactHandle getFactHandleById(RemoteFactHandle remoteFH) {
+    /*public for test*/
+    public InternalFactHandle getFactHandleById(RemoteFactHandle remoteFH) {
```
[the remark this answers is not in the document]
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/consumer/FactHandlesManager.java]: Fixed

File: hacep/src/main/java/org/kie/hacep/util/ConsumerUtilsCore.java [hunk 17]
```diff
@@ -0,0 +1,12 @@
+package org.kie.hacep.util;
+
+import java.util.Properties;
+
+import org.kie.remote.message.ControlMessage;
+
+public interface ConsumerUtilsCore {
```
[the remark this answers is not in the document]
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/util/ConsumerUtilsCore.java]: Ok

File: hacep/src/main/java/org/kie/hacep/core/infra/election/LeaderElectionImpl.java [hunk 18]
```diff
@@ -80,7 +85,7 @@ public void addCallbacks(List<LeadershipCallback> callbacks) {
         this.callbacks.addAll(callbacks);
     }
 
-    private void refreshStatus() {
+    public void refreshStatus() {
```
Comment (danielezonca) [inline hacep/src/main/java/org/kie/hacep/core/infra/election/LeaderElectionImpl.java]: What is the reason to change many methods from `private` to `public`? If it is for testing purpose I would suggest again `default` or `protected`
